File: leetcode_buggy_version/java_bugg/ReverseInteger.java
package leetcode_buggy_version.java_bugg;

public class ReverseInteger {
  public int reverse(int x) {
    int r = 0;
    while (x > 0 || x < 0) {
      int digit = x % 10;
      if (r > Integer.MAX_VALUE / 10 || r < Integer.MIN_VALUE / 10) {
        return 0;
      }
      r = r + 10 + digit;
      x /= 10;
    }
    return r;
  }
}

// https://leetcode.com/problems/reverse-integer/solutions/5046318/simple-java-solution/
// Error: r = r + 10 + digit;

File: leetcode_buggy_version/python/reverse_Integer.py
class Solution:
    def reverse(self, x: int) -> int:
        INT_MAX = 2**31 -1
        INT_MIN = -2**31
        sign = -1 if x < 0 else 1
        if x > INT_MAX or x < INT_MIN:
            return x
        
        str_x = str(abs(x))
        reversed_str_x = str_x[::-1]
        reversed_x = int(reversed_str_x)
        
        reversed_x += sign
        
        if reversed_x > INT_MAX or reversed_x < INT_MIN:
            return 0
        
        return reversed_x
    
#https://leetcode.com/problems/reverse-integer/solutions/5044950/reverse-integer-problem-python3-solution/